tests/operators/test_regrid.py

```diff
@@ -175,6 +219,21 @@ def test_regrid_to_single_point_missing_coord(cube):
         regrid.regrid_to_single_point(source, 0.5, 358.5, "Nearest")
 
 
+def test_longitude_fix_callback_missing_coord(cube):
+    """Missing coordinate raises error."""
+    # Missing X coordinate.
+    source = cube.copy()
+    source.remove_coord("grid_longitude")
+    cube_fix = read._longitude_fix_callback(source, None, None)
+    assert cube_fix == source
+
+    # Missing Y coordinate.
+    source = cube.copy()
+    source.remove_coord("grid_latitude")
+    cube_fix = read._longitude_fix_callback(source, None, None)
```

Comment:
```suggestion
    cube_fix = read._longitude_fix_callback(source.copy(), None, None)
    assert cube_fix == source

    # Missing Y coordinate.
    source = cube.copy()
    source.remove_coord("grid_latitude")
    cube_fix = read._longitude_fix_callback(source.copy(), None, None)
```
As the callback modifies the cube in place, we need to ensure it is working on a different cube, otherwise this test will never be able to fail.

Reply:
Good spot -- the ".copy()" has been added in both instances.